recirq/lattice_gauge/lattice_gauge_experiment.py

```diff
@@ -601,7 +601,7 @@ def cnot_on_layer(
             cirq.Moment(cirq.CZ.on(qc, qt) for qc, qt in pairs_list),
             cirq.Moment(cirq.H.on_each(pair[1] for pair in pairs_list)),
         ]
-    elif type(depolarization_probability) == float:
+    elif isinstance(depolarization_probability, float):
```

Comment:
(non-blocking) consider changing to `numbers.Real` if we'd like to handle np.float32 values.

Reply:
Good idea. Since there may be other places in the code where a similar change ought to be made, I won't do it in this PR, and instead I opened an issue (#435) to remind us to do a broader check for numerical type comparisons that might not behave as expected with NumPy 2.